File: concurrency_practice/src/main/java/com/yunhalee/concurrency_practice/repository/CouponRepository.java
package com.yunhalee.concurrency_practice.repository;

import com.yunhalee.concurrency_practice.domain.Coupon;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

@Repository
public interface CouponRepository extends JpaRepository<Coupon, Long> {

}

File: concurrency_redisson/src/main/kotlin/com/yunhalee/concurrency_redisson/repository/coupon/CouponRepository.kt
package com.yunhalee.concurrency_redisson.repository.coupon

import com.yunhalee.concurrency_redisson.domain.coupon.Coupon
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface CouponRepository : JpaRepository<Coupon, Long> {
    fun countAllByPromotionId(promotionId: Long): Int
}
